### PR TITLE
Fix: Misleading information about available time slots

### PR DESCRIPTION
**Pull request by vladimir-8**
Fixes: https://github.com/coopcycle/coopcycle-web/issues/4050

### CODE REVIEW

**Review by lucasferraro — COMMENTED** [review]
@vladimir-8 the code seems to be fine but I want to check in my local environment how it works as well. 
Also, Could you attach to the PR a GIF or VIDEO to see how this implementation changes the current behaviour? Thanks

**Review by Atala — APPROVED** [review]
hello i made a few comments here and there

I have not configured correctly the platform (stripe accounts) so I can't easily modify restaurants schedule to test as much as I want. I think it is good if it works for you, but if you want i can take the time to configure properly the platform on my local machine beginning of next week. otherwise i can test quickly in demo before releasing. anyway we should release to just one foodtech coop

i am under the impression this PR solves maybe 3 or 4 different issues ? it is better if you indicate this in the PR directly so we can check better when rewieving

File: js/app/restaurant/components/Order/FulfillmentDetails.js [hunk 1]
```diff
@@ -1,46 +1,72 @@
...
 
+  const [isWarningModalOpen, setWarningModalOpen] = useState(false)
+  const continueURL = window.Routing.generate('order_continue')
+
+  const { t } = useTranslation()
+
   const dispatch = useDispatch()
 
+  const changeFulfillmentMethod = () => {
+    if (canAddToExistingCart) {
```
[the remark this answers is not in the document]
Comment (vladimir-8) [reply js/app/restaurant/components/Order/FulfillmentDetails.js:41]: True. Initially, I wanted to fix https://github.com/coopcycle/coopcycle-web/issues/4081 in the same PR, but it turned out to be more complicated, so I limited this PR to time slots only

File: js/app/restaurant/components/Order/FulfillmentDetails.js [hunk 2]
```diff
@@ -1,46 +1,72 @@
...
         <div className="fulfillment-details">
           <FulfillmentMethod
             value={ fulfillmentMethod }
             shippingAddress={ cart.shippingAddress }
-            onClick={ () => dispatch(openAddressModal(cart.restaurant)) }
-            allowEdit={ !isPlayer } />
-          { isOrderingAvailable && <Time /> }
+            onClick={ changeFulfillmentMethod }
+            allowEdit={ isOrderAdmin } />
+          { Boolean(fulfilmentTimeRange) ? (
```
[the remark this answers is not in the document]
Comment (vladimir-8) [reply js/app/restaurant/components/Order/FulfillmentDetails.js:65]: That's British English 😄 https://www.grammarly.com/blog/fulfil-fulfill/

File: js/app/restaurant/list.js [hunk 3]
```diff
@@ -32,7 +34,11 @@ import './list.scss'
  */
 window.history.scrollRestoration = 'manual'
 
-const FulfillmentBadge = ({ range, isPreOrder }) => {
+const FulfillmentBadge = ({ fulfilmentMethod, isPreOrder }) => {
+
+  if (!fulfilmentMethod) {
```
[the remark this answers is not in the document]
Comment (vladimir-8) [reply js/app/restaurant/list.js:39]: Also true, it's a quick fix for: https://github.com/coopcycle/coopcycle-app/pull/1689#issuecomment-1994390026

**Pull request merged**